pkg/template/processor.go

```diff
@@ -47,13 +49,54 @@ func apply(cl client.Client, objs []runtime.RawExtension) error {
 			continue
 		}
 		gvk := obj.GetObjectKind().GroupVersionKind()
-		if err := createOrUpdateObj(cl, obj); err != nil {
+		if err := createOrUpdateObj(p.cl, obj); err != nil {
 			return errs.Wrapf(err, "unable to create resource of kind: %s, version: %s", gvk.Kind, gvk.Version)
 		}
+		if gvk.Group == "project.openshift.io" && gvk.Version == "v1" && gvk.Kind == "ProjectRequest" {
+			if _, ok := obj.(runtime.Unstructured); ok {
+				var prq projectv1.ProjectRequest
+				err := p.scheme.Convert(obj, &prq, nil)
+				if err != nil {
+					return err
+				}
+				// wait until the Project exists and is ready
+				_, err = p.waitUntilProjectExists(prq.GetName(), 2*time.Second)
```

Comment:
`2 sec` timeout may be less any reference on arriving this number ??  anyway, we are polling on some interval .. so we can safely increase timeout.

Reply:
ok, I suggest that we use a higher, hard-coded value for now (10 sec?) and work on making this configurable in a subsequent PR. Ok for you?

Reply:
value set to 10s in 92a2994